cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -106,8 +110,27 @@ class GcsTestbench : public ::testing::Environment {
         continue;
       }
 
-      server_process_ = bp::child(exe_path, "-m", "testbench", "--port", port_, group_);
-      if (server_process_.valid() && server_process_.running()) break;
+      bp::ipstream output;
```

Comment:
This still makes me suspicious since server_process_ outlives output. Maybe to make this a member variable, it has to be re-initialized on each loop iteration?

Reply:
I've tried several different ways, but I still have no idea how to make the output a member variable without making the unit tests freeze up. 🤷

Reply:
Hmm. Weird. Do you at least have a stacktrace or something…?

Reply:
Yeah it hangs within SetUp on CreateBucket. For some reason the testbench server becomes unresponsive.

Reply:
<details>
__select (@__select:5)
Curl_poll (@Curl_poll:150)
multi_wait (@multi_wait:201)
curl_easy_perform (@curl_easy_perform:69)
google::cloud::storage::v1_42_0::internal::CurlHandle::EasyPerform() (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/storage/internal/curl_handle.h:98)
google::cloud::storage::v1_42_0::internal::CurlRequest::MakeRequestImpl() (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/storage/internal/curl_request.cc:129)
google::cloud::storage::v1_42_0::internal::CurlRequest::MakeRequest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) && (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/storage/internal/curl_request.cc:77)
google::cloud::storage::v1_42_0::internal::CurlClient::CreateBucket(google::cloud::storage::v1_42_0::internal::CreateBucketRequest const&) (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/storage/internal/curl_client.cc:209)
google::cloud::storage::v1_42_0::internal::raw_client_wrapper_utils::Signature<google::cloud::v1_42_0::StatusOr<google::cloud::storage::v1_42_0::BucketMetadata> (google::cloud::storage::v1_42_0::internal::RawClient::*)(google::cloud::storage::v1_42_0::internal::CreateBucketRequest const&)>::ReturnType google::cloud::storage::v1_42_0::internal::(anonymous namespace)::MakeCall<google::cloud::v1_42_0::StatusOr<google::cloud::storage::v1_42_0::BucketMetadata> (google::cloud::storage::v1_42_0::internal::RawClient::*)(google::cloud::storage::v1_42_0::internal::CreateBucketRequest const&)>(google::cloud::v1_42_0::internal::TraitBasedRetryPolicy<google::cloud::storage::v1_42_0::internal::StatusTraits>&, google::cloud::v1_42_0::internal::BackoffPolicy&, google::cloud::v1_42_0::Idempotency, google::cloud::storage::v1_42_0::internal::RawClient&, google::cloud::v1_42_0::StatusOr<google::cloud::storage::v1_42_0::BucketMetadata> (google::cloud::storage::v1_42_0::internal::RawClient::*)(google::cloud::storage::v1_42_0::internal::CreateBucketRequest const&), google::cloud::storage::v1_42_0::internal::raw_client_wrapper_utils::Signature<google::cloud::v1_42_0::StatusOr<google::cloud::storage::v1_42_0::BucketMetadata> (google::cloud::storage::v1_42_0::internal::RawClient::*)(google::cloud::storage::v1_42_0::internal::CreateBucketRequest const&)>::RequestType const&, char const*) (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/storage/internal/retry_client.cc:62)
google::cloud::storage::v1_42_0::internal::RetryClient::CreateBucket(google::cloud::storage::v1_42_0::internal::CreateBucketRequest const&) (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-prefix/src/google_cloud_cpp_ep/google/cloud/storage/internal/retry_client.cc:132)
google::cloud::v1_42_0::StatusOr<google::cloud::storage::v1_42_0::BucketMetadata> google::cloud::storage::v1_42_0::Client::CreateBucketForProject<>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, google::cloud::storage::v1_42_0::BucketMetadata) (/Users/willjones/Documents/arrows/arrow/cpp/build/ninja-debug-python-user/google_cloud_cpp_ep-install/include/google/cloud/storage/client.h:433)
arrow::fs::(anonymous namespace)::GcsIntegrationTest::SetUp() (/Users/willjones/Documents/arrows/arrow/cpp/src/arrow/filesystem/gcsfs_test.cc:192)
arrow::fs::(anonymous namespace)::TestGCSFSGeneric::SetUp() (/Users/willjones/Documents/arrows/arrow/cpp/src/arrow/filesystem/gcsfs_test.cc:305)
void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (@void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*):28)
testing::Test::Run() (@testing::Test::Run():29)
testing::TestInfo::Run() (@testing::TestInfo::Run():62)
testing::TestSuite::Run() (@testing::TestSuite::Run():92)
testing::internal::UnitTestImpl::RunAllTests() (@testing::internal::UnitTestImpl::RunAllTests():227)
bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (@bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*):28)
testing::UnitTest::Run() (@testing::UnitTest::Run():34)
main (@main:20)
start (@start:133)
</details>

Reply:
It seems the server is mostly OK if you explicitly close stdout, from a quick test in Python. So I guess things are OK here.